cartesia/_async_websocket.py

```diff
@@ -105,6 +108,95 @@ async def no_more_inputs(self) -> None:
             context_id=self._context_id,
             continue_=False,
         )
+    
+    async def flush(self) -> Callable[[], AsyncGenerator[Dict[str, Any], None]]:
+        """Trigger a manual flush for the current context's generation. This method returns a generation that yields the audio prior to the flush."""
+        await self.send(
+            model_id=DEFAULT_MODEL_ID,
+            transcript="",
+            output_format=TTS.get_output_format("raw_pcm_f32le_44100"),
+            voice_embedding=DEFAULT_VOICE_EMBEDDING,  # Default voice embedding since it's a required input for now.
+            context_id=self._context_id,
+            continue_=True,
+            flush=True,
+        )
+
+        # Save the old flush ID
+        flush_id = len(self._websocket._context_queues[self._context_id]) - 1
+
+        # Create a new Async Queue to store the responses for the new flush ID
+        self._websocket._context_queues[self._context_id].append(asyncio.Queue())
+
+        # Return the generator for the old flush ID
+        async def generator():
+            try:
+                while True:
+                    # Create tasks for current queue
+                    current_queue_task = asyncio.create_task(self._websocket._get_message(self._context_id, timeout=self.timeout, flush_id=flush_id))
+                    tasks = [current_queue_task]
+                    
+                    # If next queue exists, create a task to check it
+                    next_queue_task = asyncio.create_task(self._websocket._get_message(self._context_id, timeout=self.timeout, flush_id=flush_id + 1))
+                    tasks.append(next_queue_task)
+
+                    # Wait for either:
+                    # 1. Current queue to get a message
+                    # 2. Next queue to get a message while current queue is empty
+                    # 3. Timeout to occur (only if next queue is empty)
+                    # Print the length of each queue
+                    done, pending = await asyncio.wait(
+                        tasks,
+                        return_when=asyncio.FIRST_COMPLETED,
+                        timeout=self.timeout
+                    )
+
+                    completed_or_cancelled_tasks = [completed_task for completed_task in done]
```

Comment:
nit: `cancelled` -> `canceled` (all instances) [`cancelled` is British spelling]

Reply:
I'm a big fan of the Queen's English